spec/integration/schema/macros/value_spec.rb

```diff
@@ -217,5 +217,24 @@
         expect(schema.(data: { num: '11' })).to be_success
       end
     end
+
+    context 'with a hash type' do
+      let(:type) do
+        Types::Hash.schema(name: 'string')
+      end
+
+      subject(:schema) do
+        env = self
+        Dry::Schema.define do
+          required(:data).value(env.type)
+        end
+      end
+
+      it 'unpacks type' do
+        expect(schema.(data: { name: nil }).messages).to eql(
+          data: { name: ['must be a string'] }
+        )
+      end
+    end
```

Comment:
Could you add an example with a deeply nested hash?

Reply:
@solnic I'll be adding all sorts of examples, stay tuned